pynetsnmp/netsnmp.py

```diff
@@ -157,7 +157,24 @@ class netsnmp_transport(Structure):
     pass
 
 
-# int (*netsnmp_callback) (int, netsnmp_session *, int, netsnmp_pdu *, void *);
+# include/net-snmp/types.h
+class netsnmp_trap_stats(Structure):
+    _fields_ = [
+        ("sent_count", c_ulong),
+        ("sent_last_sent", c_ulong),
+        ("sent_fail_count", c_ulong),
+        ("sent_last_fail", c_ulong),
+        ("ack_count", c_ulong),
+        ("ack_last_rcvd", c_ulong),
+        ("sec_err_count", c_ulong),
+        ("sec_err_last", c_ulong),
+        ("timeouts", c_ulong),
+        ("sent_last_timeout", c_ulong),
+    ]
+
+
+# include/net-snmp/types.h -> int (*netsnmp_callback) (int, netsnmp_session *, int, netsnmp_pdu *, void *);
+# the first argument is self. Without it, callback function throws an error with not enough args, 4 from 5
```

Comment:
Your comment here made me take a closer look at the this code to try to understand what you meant "the first argument is self."

It turns out that this declaration does _not_ match the function it's being mapped to on line 611 (the actual mapping occurs on line 626).  The function declared on line 611 takes five arguments which matches what the `netsnmp_callback` function is declared with in the net-snmp/types.h header file.

Can you provide some more details on what's happening here?  I made a change to accommodate what looked like a memory overwriting situation in the arguments passed to the `netsnmp_callback` function, but I missed that the declaration had an extra `int` at the beginning of its argument list.  Maybe removing this extra `int` will make the work-around I came up with (line 494) unnecessary?

Reply:
The ticket where I recorded my investigation is https://zenoss.atlassian.net/browse/ZEN-29805.

It could be that the extra `int` in the ctypes declaration is irrelevant because I was able to reproduce the issue using pure C code (found in the ticket).